### PR TITLE
Redesign Create Comment Screen

### PR DESCRIPTION
**Pull request by coslu**
## Pull Request Description

This PR redesigns the UI of the create comment screen and adds some additional improvements to the create post screen.

* The create comment screen is no longer displayed as a bottom modal sheet. It is now a page that matches the design of the create post page. 
* Allows to upload images in comments. All image buttons in the create post- and comment pages turn into a `CircularProgressIndicator` while an image is being uploaded.
* When replying directly to a post, the contents of the post are shown in the 'replying to' area. 
* When the create comment page is opened, the text field has focus and the keyboard opens automatically. 
* When switching back from preview, the previous cursor location is maintained and the text field gains focus. In create post page, body text field gains focus.
* Create post- and comment pages show an indicator for the currently active account with avatar, display name, user name and instance.
* Prevents opening the create post- and comment pages when not logged in (was possible through FAB).

## Issue Being Fixed

Issue Number: N/A

## Screenshots / Recordings

Screenshot_20230815_184921 

## Checklist

- [x] Did you update CHANGELOG.md?
- [x] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/community/pages/create_post_page.dart
Comment (micahmo) [inline lib/community/pages/create_post_page.dart:112]: This is probably a question that @hjiangsu can answer better, but are we mixing styles here by calling `setState` from a `BlocListener`? I believe the builder will be called again when the state changes, so can we check `state.status` in the `IconButton` rather than using a separate property and rebuilding again?

File: lib/post/pages/create_comment_page.dart
Comment (micahmo) [inline lib/post/pages/create_comment_page.dart:164]: This snackbar will appear for a whole day?? 

Also a quick note about all snackbars, I'm working on a refactor in #571, so depending on who goes first, I'll include this in my changes or vice versa!

File: lib/post/pages/post_page.dart
Comment (micahmo) [inline lib/post/pages/post_page.dart:193]: This should already be in `develop`, so once you update your branch, hopefully there will be fewer changes here.

https://github.com/thunder-app/thunder/blob/develop/lib/post/pages/post_page.dart#L191

File: lib/utils/image.dart
Comment (micahmo) [inline lib/utils/image.dart:138]: Do we want to propagate up some error handling?

**Review by micahmo — COMMENTED** [review]
Looks lovely!

File: lib/post/pages/create_comment_page.dart
Comment (coslu) [reply lib/post/pages/create_comment_page.dart:164]: Right, there I was just trying to make it stay indefinitely until the user gets rid of it. I guess it might not be needed. Should I just remove the duration? I am also aware of the refactor, I can change my snackbars once that is merged.

Comment (micahmo) [reply lib/post/pages/create_comment_page.dart:164]: Ahh I see what you're going for. Currently all of the snackbars auto-dismiss, so this would be a new pattern. I might not even think to dismiss it unless it had an `x` button. But of course I'm totally open to a new approach. I'd be interested to hear from others.

File: lib/community/pages/create_post_page.dart
Comment (hjiangsu) [reply lib/community/pages/create_post_page.dart:112]: Hmm, if we are only using BlocListener, I don't know if we have access to the state from the child. However, if we were using a BlocBuilder or BlocConsumer, then I would agree with this.

I think its okay to set states in here given that we're not using BlocBuilder/BlocConsumer

File: lib/post/pages/create_comment_page.dart
Comment (hjiangsu) [reply lib/post/pages/create_comment_page.dart:164]: The refactor should now be merged in! I think we can just set a long duration on the snackbars for now - enough time so that the message can be read or a screenshot can be taken

**Review by hjiangsu — COMMENTED** [review]
First of all, great job on this @coslu! This is a pretty big change so thanks for taking it on 🗡️ 

I just added some comments to some of the existing comments to provide my thoughts. Overall, I think this is pretty good, and should be good to go once the comments are addressed/conflicts are fixed!

**Review by micahmo — APPROVED** [review]
Since you addressed the comments and fixed the merge conflicts, I'm going to go ahead and merge this so people can start using it. Thanks again!

**Pull request merged**
